spec/PeriodicBackgroundSync-index.bs

```diff
@@ -92,3 +94,95 @@ Fetch requests within the <code>periodicsync</code> event while [=in the backgro
+        1. Else, set <var>timeTillNextAllowedPeriodicSync</var> to |minIntervalAcrossOrigins| - |timeSinceLastPeriodicSync|. Set |timeToFire| to the maximum of |delayForOrigin| + |now|, and |timeTillNextAllowedPeriodicSync| + |now|.
+    1. Set the [=time to fire=] of |registration| to |timeToFire|.
+1. Else:
+    1. Let <var>countRetries</var> be the count of retries that have been made for this event.
+    1. Increment |countRetries|.
+    1. If |countRetries| is equal to the [=maximum number of retries=] allowed, set |countRetries| to 0 and follow the steps to [=first attempt of firing|fire the first attempt=].
+    1. Else, Set the [=time to fire=] of |registration| to |now| + a small back-off that is exponentially proportional to |countRetries|.
+
+API Description {#api-description}
+===================================
+## Extensions to the {{ServiceWorkerGlobalScope}} interface ## {#extensions-to-serviceworkerglobalscope}
+<script type="idl">
+partial interface ServiceWorkerGlobalScope {
+    attribute EventHandler onperiodicsync;
+};
+</script>
+
+## Extensions to the {{ServiceWorkerRegistration}} interface ## {#extensions-to-serviceworkerregistration}
+
+<script type="idl">
+[Exposed=(Window,Worker)]
+partial interface ServiceWorkerRegistration {
+  readonly attribute PeriodicSyncManager periodicSync;
+};
+</script>
+
+<div dfn-for="ServiceWorkerRegistration">
+A {{ServiceWorkerRegistration}} has a <dfn>periodic sync manager</dfn> (a {{PeriodicSyncManager}}), initially a new {{PeriodicSyncManager}} whose [=service worker registration=] is the [=context object=]'s [=service worker registration=].
+
+The <dfn attribute>periodicSync</dfn> attribute's getter must return the [=context object=]'s [=ServiceWorkerRegistration/periodic sync manager=].
+</div>
+
+## {{PeriodicSyncManager}} interface ## {#periodicsyncmanager-interface}
+<script type="idl">
+[Exposed=(Window,Worker)]
+interface PeriodicSyncManager {
+    Promise<void> register(DOMString tag, optional BackgroundSyncOptions options);
+    Promise<sequence<DOMString>> getTags();
+    Promise<void> unregister(DOMString tag);
+};
+
+dictionary BackgroundSyncOptions {
+    [EnforceRange] unsigned long long minInterval = 0;
+};
+</script>
+
+The <code><dfn method for=PeriodicSyncManager title="register(tag, options)">register(<var>tag</var>, <var>options</var>)</dfn></code> method, when invoked, MUST return <a>a new promise</a> <var>promise</var> and run the following steps <a>in parallel</a>:
+
+1. Let <var>serviceWorkerRegistration</var> be the {{PeriodicSyncManager}}'s associated [=service worker registration=].
+1. If |serviceWorkerRegistration|’s [=active worker=] is null, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
+1. if the {{PermissionState}} for a {{PermissionDescriptor}} with {{name}} <code>"periodic-background-sync"</code> is not {{granted}}, [=reject=] |promise| with a {{NotAllowedError}} and abort these steps.
```

Comment:
does {{name}} point to anything here? It should be scoped with the outer struct

Reply:
it points to this: https://w3c.github.io/permissions/#dom-permissiondescriptor-name

Reply:
I don't know what bikeshed magic makes this work, but let's future-proof it by using {{PermissionDescriptor/name}}

Reply:
Will do. 
This is what makes this work:
https://tabatkins.github.io/bikeshed/#autolink-wrappers

Also, there's no need to futureproof, btw, because bikeshed throws an error when it comes across multiple definitions of the same name.

Reply:
Yes, but `name` is an overloaded term. In the future it might end up pointing to a different spec. From my experience it doesn't always point to the right thing, it has a list of predefined specs it might default to.

Reply:
ok, thanks for explaining that. I've already updated it.